src/meta/typeutils.go

```diff
@@ -120,204 +120,205 @@ func wrap(typ byte, byteFields []uint8, args ...string) string {
 	return string(buf)
 }
 
-func unwrap1(s string) (one string) {
-	return s[stringLenBytes+1:] // do not care about length, there is only 1 param
+func (t Type) unwrap1() (one string) {
+	return t.String()[stringLenBytes+1:] // do not care about length, there is only 1 param
 }
 
-func unwrap2(s string) (one, two string) {
+func (t Type) unwrap2() (one, two string) {
 	var l int
 	var b [stringLenBytes]byte
 	var rawBuf [stringLenBytes / 2]byte
 
 	pos := 1
-	copy(b[:], s[pos:pos+stringLenBytes])
+	copy(b[:], t.String()[pos:pos+stringLenBytes])
 	hex.Decode(rawBuf[:], b[:])
 	l = int(binary.LittleEndian.Uint16(rawBuf[:]))
 	pos += stringLenBytes
-	one = s[pos : pos+l]
+	one = t.String()[pos : pos+l]
 	pos += l
-	two = s[pos+stringLenBytes:] // do not care about length of last param
+	two = t.String()[pos+stringLenBytes:] // do not care about length of last param
 
 	return one, two
 }
 
-func unwrap3(s string) (b1 uint8, one, two string) {
+func (t Type) unwrap3() (b1 uint8, one, two string) {
 	var l int
 	var b [stringLenBytes]byte
 	var rawBuf [stringLenBytes / 2]byte
 
 	pos := 1
-	copy(b[:], s[pos:pos+uint8fieldBytes])
+	copy(b[:], t.String()[pos:pos+uint8fieldBytes])
```

Comment:
Not sure if you even need to call String() here to be honest because the underlying type is already string